tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -6055,6 +6082,19 @@ SpirvEmitter::tryToAssignToMatrixElements(const Expr *lhs,
   return rhs;
 }
 
+SpirvInstruction *SpirvEmitter::tryToAssignToStringVar(const Expr *lhs,
+                                                       SpirvInstruction *rhs) {
+  // Assigning to a 'string' variable. Since SPIR-V does not have a 'string'
+  // type, assigning a string literal to a string variable is only a
+  // matter of updating the literal that is registered in DeclResultIdMapper.
+  if (isStringType(lhs->getType()))
+    if (auto *lhsDeclExpr = dyn_cast<DeclRefExpr>(lhs))
+      if (auto *varDecl = dyn_cast<VarDecl>(lhsDeclExpr->getDecl()))
+        return declIdMapper.createOrUpdateStringVar(varDecl, rhs);
```

Comment:
I am worried the following case:

```
string a = "a"
if (... ) { ... a = "b"; ... }
printf(a);
```

As far as I understand, the value of `a` will be always "b" even though the condition of if-statement is false.

If the answer is yes, do you have any idea how to solve this issue?
(I am not sure this is possible to solve .. because SPIR-V does not have `OpVariable` for string type.. right?)

Reply:
Yeah, I agree.